tests/processor/test_tasks.py

```diff
@@ -109,31 +111,43 @@ class TestProcessorTasks(MasuTestCase):
 
     fake = faker.Faker()
 
+    def setUp(self):
+        super().setUp()
+
+        self.fake_reports = []
+        for _ in range(1, random.randint(5,20)):
+            self.fake_reports.append({'file': self.fake.word(),
+                                      'compression': random.choice(['GZIP', 'PLAIN'])})
+
+        fake_account = fake_arn(service='iam', generate_account_id=True)
+
+        self.fake_get_report_args = {'customer_name': self.fake.word(),
+                                    'authentication': fake_account,
+                                    'provider_type': 'AWS',
+                                    'report_name': self.fake.word(),
+                                    'schema_name': self.fake.word(),
+                                    'billing_source': self.fake.word(),
+                                    'provider_id': random.randint(1,65535)}
+
+        self.today = datetime.today()
+        self.yesterday = datetime.today() - timedelta(days=1)
+
     @patch('masu.processor.tasks._get_report_files')
     @patch('masu.processor.tasks.process_report_file')
     def test_get_report_files_second_task_called(self,
                                                  mock_process_files,
                                                  mock_get_files):
         """Test that the chained task is called."""
-        report_dicts = [{'file': 'file1', 'compression': 'GZIP'},
-                        {'file': 'file2', 'compression': 'PLAIN'}]
-
         mock_process_files.delay = Mock()
-        mock_get_files.return_value = report_dicts
+        mock_get_files.return_value = self.fake_reports
 
-        account = fake_arn(service='iam', generate_account_id=True)
-        schema_name = self.fake.word()
-        get_report_files(customer_name=self.fake.word(),
-                         authentication=account,
-                         provider_type='AWS',
-                         report_name=self.fake.word(),
-                         schema_name=schema_name,
-                         billing_source=self.fake.word(),
-                         provider_id=random.randint(1,65535))
+        get_report_files(**self.fake_get_report_args)
```

Comment:
I know it's easier to write, and in test code, I don't mind the shortcut, really, but to expand on my point on `**kwargs` and readability, here now the reader has to scroll through the file to find the dictionary that is getting passed in to mentally unpack it into the function arguments. The get_report_files function has well defined parameters, so we're not gaining anything by the the use of dictionary unpacking.

Reply:
In this case, it was more of a DRY issue. We were using the same param values across multiple tests.

Where I'm at right now in my learning a more TDD workflow, I find it really helpful to keep the test function itself as clean as possible, so that I can clearly see what's being tested and what pieces are changing from test to test.